components/time_picker/Clock.jsx

```diff
@@ -30,7 +30,9 @@ class Clock extends React.Component {
 
   componentDidMount () {
     window.addEventListener('resize', this.handleCalculateShape);
-    this.handleCalculateShape();
+    setTimeout(() => {
+      this.handleCalculateShape();
+    });
```

Comment:
I am not sure if this could ever happen, but what if this component get unmounted again before the `timeout` trigger is fired?

you might need to do something like this:
```JSX
componentDidMount() {
    ...
    this._cancelTimeout = setTimeout(() => this.handleCalculateShape());
}
componentWillUnmount() {
    ...
    cancelTimeout(this._cancelTimeout);
    this._cancelTimeout = undefined;
}
```
@Blechhirn